spec.html

```diff
@@ -18,6 +18,102 @@ <h1>Array Objects</h1>
+                    1. If _comparefn_ is not *undefined*, then
+                        1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
+                        1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
+                        1. Return _v_.
+                    1. [id="step-sortcompare-tostring-x"] Let _xString_ be ? ToString(_x_).
+                    1. [id="step-sortcompare-tostring-y"] Let _yString_ be ? ToString(_y_).
+                    1. Let _xSmaller_ be ! IsLessThan(_xString_, _yString_, *true*).
+                    1. If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.
+                    1. Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).
+                    1. If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.
+                    1. Return *+0*<sub>𝔽</sub>.
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-sortindexedproperties" type="abstract operation">
+                <h1>
+                  SortIndexedProperties (
+                    _obj_: an Object,
+                    <ins>_out_: an Object,</ins>
+                    _len_: a non-negative integer,
+                    _SortCompare_: an Abstract Closure with two parameters,
+                    <ins>_checkHasProperty_: a Boolean</ins>
+                  ): either a normal completion containing an Object or an abrupt completion
+                </h1>
+                <dl class="header">
+                </dl>
+                <emu-alg>
+                  1. Let _items_ be a new empty List.
+                  1. Let _k_ be 0.
+                  1. Repeat, while _k_ &lt; _len_,
+                    1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                    1. <del>Let _kPresent_ be ? HasProperty(_obj_, _Pk_).</del>
+                    1. <ins>If _checkHasProperty_ is *true*, then</ins>
+                        1. <ins>Let _kRead_ be ? HasProperty(_obj_, _Pk_).</ins>
+                    1. <ins>Else,</ins>
+                        1. <ins>Let _kRead_ be *true*.</ins>
+                    1. If <del>_kPresent_</del><ins>_kRead_</ins> is *true*, then
+                      1. Let _kValue_ be ? Get(_obj_, _Pk_).
+                      1. Append _kValue_ to _items_.
+                    1. Set _k_ to _k_ + 1.
+                  1. Let _itemCount_ be the number of elements in _items_.
+                  1. [id="step-array-sort"] Sort _items_ using an implementation-defined sequence of calls to _SortCompare_. If any such call returns an abrupt completion, stop before performing any further calls to _SortCompare_ or steps in this algorithm and return that Completion Record.
+                  1. Let _j_ be 0.
+                  1. Repeat, while _j_ &lt; _itemCount_,
+                    1. <del>Perform ? Set(_obj_, ! ToString(𝔽(_j_)), _items_[_j_], *true*).</del>
+                    1. <ins>Perform ? Set(_out_, ! ToString(𝔽(_j_)), _items_[_j_], *true*).</ins>
```

Comment:
We can highlight that we are only changing its first param.
```suggestion
                    1. Perform ? Set(<del>_obj_</del><ins>_out_</ins>, ! ToString(𝔽(_j_)), _items_[_j_], *true*).
```